Feature regression tests

avalanche123 opened the pull request.

Commit:
test: Adding C* profiles and ensuring CCM instances are terminated

This commit adds three C* profiles for launching feature tests against
different C* versions. Filters/Tags are also implemented to disable
tests based on unsupported features for a given native protocol version.
Before and after suite clean up function will remove all running CCM
cluster instances.

Commit:
fix: Adding missing linking library for Vagrant instance

Commit:
Adding INI information to display for --info or phpinfo()

Commit:
test: Adding basic logging feature example

avalanche123 commented on Vagrantfile:
```diff
@@ -42,7 +42,7 @@ Vagrant.configure("2") do |config|
   pushd /usr/local/src/php-driver/ext
   phpize
-  LDFLAGS="-L$builddir/lib" LIBS="-lssl -lz -luv -lgmp -lstdc++" ./configure --with-cassandra=/tmp/php-driver-installation/build --with-gmp=/tmp/php-driver-installation/build --with-libdir=lib
+  LDFLAGS="-L$builddir/lib" LIBS="-lssl -lz -luv -lm -lgmp -lstdc++" ./configure --with-cassandra=/tmp/php-driver-installation/build --with-gmp=/tmp/php-driver-installation/build --with-libdir=lib
```
why do we depend on -lm?

mikefero replied:
Logarithmic functions are being used (https://github.com/datastax/php-driver/blob/master/ext/util/math.c#L427) and were not being linked in (e.g. undefined references)

avalanche123 replied:
gotcha, thanks!

avalanche123 commented on support/ccm.php:
```diff
@@ -94,16 +94,22 @@ public function stop()
-        foreach (explode("\n", $this->run('list')) as $cluster) {
-            if ($this->isActive($cluster)) {
-                $active = count($clusters);
+        foreach (explode("\r\n", $this->run('list')) as $cluster) {
```
let's use `PHP_EOL` instead of `"\r\n"` here.

mikefero replied:
Good catch

avalanche123 commented on features/bootstrap/FeatureContext.php:
```diff
@@ -155,6 +157,41 @@ public function itsOutputShouldContain(PyStringNode $string)
+     */
+    public function theFollowingLoggerSettings(PyStringNode $string)
+    {
+        $lines = preg_split("/\\r\\n|\\r|\\n/", $string->getRaw());
```
you can use `explode(PHP_EOL, $string->getRaw())` here

mikefero replied:
Tried that and the result was one line returned from explode when on Windows platform. I can continue to play around with other options.

avalanche123 replied:
nah, leave whatever works in

Commit:
fix log level, warn when log file doesn't exist

Commit:
test: Applying logger fixes to feature test

Commit:
Merge branch 'master' into feature_regression_tests

Commit:
Merge branch 'feature_regression_tests' of github.com:datastax/php-driver into feature_regression_tests

Commit:
Correcting replication factors for single node feature tests

Commit:
Removing log message for missing log file

Commit:
test: Adding consistency level feature

This feature test adds the ability to
 * Create a multi-node and multi-datacenter cluster
 * Enable/Disable system traces via nodetool
 * Operate against multiple cluster names for feature tests

Commit:
fix: Correcting cluster create/stop/switch logic

Commit:
fix: Windows build and Behat tests against PHP v5.3

Commit:
Downgrading C* v2.0.x version until CCM fix for Windows

Commit:
test: Persistent sessions WIP

Currently a work in progress. The built in PHP web server is started
and can execute requests, but validation is still missing from
feature test.

Commit:
Adding a skip mechanism for TravisCI around PHP v5.3

Commit:
test: Adding multiple sessions object feature

Commit:
update persistent sessions feature

Commit:
use 127.0.0.1 instead of localhost

Commit:
add some retry logic when fetching a page

Commit:
short-circuit session init success

Commit:
print retry attempts

Commit:
print web server output on failure

Commit:
Adding a new filter/tag for skipping tests during TravisCI
